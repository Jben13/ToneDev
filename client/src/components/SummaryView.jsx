import React from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
// import { response } from 'spdy';
import PieChart from './PieChart';


class Summary extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      score: 73,
      sentiment: 'positive',
      desciption: '',
    };
    this.getAnalysis = this.getAnalysis.bind(this);
  }

  componentDidMount() {
    const { orgName } = this.props;
    this.getAnalysis(orgName);
  }

  getAnalysis(orgName) {
    axios.get('/api/gateway/github/orgdata', {
      orgName,
    })
      .then((response) => {
        console.log(response);
        this.setState({
          description: response.data.orgDescription,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  render() {
    const { orgName } = this.props;
    const { sentiment, score } = this.state;
    return (
      <div>
        <h1>
          {orgName}
          {' '}
        Summary
        </h1>
        <p>
        The
          {' '}
          <a href="https://freeCodeCamp.org">freeCodeCamp</a>
          {' '}
        open source codebase and curriculum.
        Learn to code for free together with millions of people.
        </p>
        <h2>
Overall Sentiment
          {' '}
          {orgName}
        </h2>
        <p>{sentiment}</p>

        <h2>Score</h2>
        <p>
          {score}
        </p>

        <PieChart score={73} sentiment="positive" />

      </div>
    );
  }
}

Summary.propTypes = {
  orgName: PropTypes.string.isRequired,
};
export default Summary;
