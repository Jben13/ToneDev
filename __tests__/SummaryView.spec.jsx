import React from 'react';
import SummaryView from '../client/src/components/SummaryView';

test('App renders component correctly', () => {
  const component = global.shallow(<SummaryView />);
  expect(component).toMatchSnapshot();
});
