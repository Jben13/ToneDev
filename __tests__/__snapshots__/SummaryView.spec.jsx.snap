// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<div>
  <h1>
    Sentiment Analysis Summary
  </h1>
  <h2>
    Overall Sentiment
  </h2>
  <p>
    positive
  </p>
  <h2>
    Score
  </h2>
  <p>
    80% positive
  </p>
  <PieChart />
  <h2>
    Keywords
  </h2>
  <p>
    great, awesome, thanks!
  </p>
</div>
`;
